t/local_package.t

```diff
@@ -67,44 +68,66 @@ subtest "constructor tests" => sub {
 };
 
 subtest 'refresh and unpack index files' => sub {
+    plan skip_all => "IO::Uncompress::Gunzip is not available"
+      unless $CPAN::Common::Index::Mirror::HAS_IO_UNCOMPRESS_GUNZIP;
     my $index = new_local_index;
 
-    ok( !-e File::Spec->catfile( $cache, $packages ), "$packages not in cache" );
+    ok( !-e File::Spec->catfile( $index->cache, $packages ), "$packages not in cache" );
 
     ok( $index->refresh_index, "refreshed index" );
 
-    ok( -e File::Spec->catfile( $cache, $packages ), "$packages in cache" );
+    ok( -e File::Spec->catfile( $index->cache, $packages ), "$packages in cache" );
 };
 
 subtest 'refresh and unpack uncompressed index files' => sub {
     my $index = new_uncompressed_local_index;
 
-    ok( !-e File::Spec->catfile( $cache, $uncompressed ), "$uncompressed not in cache" );
+    ok( !-e File::Spec->catfile( $index->cache, $uncompressed ),
+        "$uncompressed not in cache" );
 
     ok( $index->refresh_index, "refreshed index" );
 
-    ok( -e File::Spec->catfile( $cache, $uncompressed ), "$uncompressed in cache" );
+    ok( -e File::Spec->catfile( $index->cache, $uncompressed ),
+        "$uncompressed in cache" );
 };
 
 # XXX test that files in cache aren't overwritten?
 
-subtest 'check index age' => sub {
-    my $index   = new_local_index;
-    my $package = $index->cached_package;
-    ok( -f $package, "got the package file" );
-    my $expected_age = ( stat($package) )[9];
-    is( $index->index_age, $expected_age, "index_age() is correct" );
+my $common_tests = sub {
```

Comment:
I don't see the benefit of making this an anonymous subroutine.  Doing so makes me search for some reason why it's not a normal subroutine, so I'd prefer to have it just be a  normal sub (unless there is some reason I'm missing).